src/SBarcodeScanner.cpp

```diff
@@ -0,0 +1,121 @@
+#include "SBarcodeScanner.h"
+
+SBarcodeScanner::SBarcodeScanner(QObject *parent)
+    : QVideoSink(parent)
+    ,	camera(nullptr) {
+
+    connect(&m_decoder, &SBarcodeDecoder::capturedChanged, this, &SBarcodeScanner::setCaptured);
+    connect(this, &QVideoSink::videoFrameChanged, this, &SBarcodeScanner::handleFrameCaptured);
+
+    worker = new Worker(this);
+    worker->moveToThread(&workerThread);
+    connect(&workerThread, &QThread::finished, worker, &QObject::deleteLater);
+    connect(this, &SBarcodeScanner::process, worker, &Worker::process);
+    workerThread.start();
+
+    initCam();
+}
+
+SBarcodeScanner::~SBarcodeScanner()
+{
+    workerThread.quit();
+    workerThread.wait();
+    stopCam();
+}
+
+SBarcodeDecoder *SBarcodeScanner::getDecoder()
+{
+    return &m_decoder;
+}
+
+void SBarcodeScanner::initCam() {
+    camera = new QCamera(this);
+
+    const auto settings = camera->cameraDevice().videoFormats();
+
+    int i = 0;
+    const auto s = settings.at(i);
+
+    camera->setFocusMode(QCamera::FocusModeAuto);
+    camera->setCameraFormat(s);
+
+    m_capture.setCamera(camera);
+    m_capture.setVideoSink(this);
+
+    camera->start();
+}
+
+void SBarcodeScanner::stopCam()
+{
+    camera->stop();
+    disconnect(camera, 0, 0, 0);
+    camera->setParent(nullptr);
+    delete camera;
+    camera = nullptr;
+}
+
+void SBarcodeScanner::handleFrameCaptured(const QVideoFrame &frame) {
+
+    emit process(frame.toImage().convertToFormat(QImage::Format_ARGB32));
+
+    if(m_videoSink) {
+        m_videoSink->setVideoFrame(frame);
+    }
+
+    pauseProcessing();
+}
+
+void SBarcodeScanner::imageProcess(SBarcodeDecoder *decoder, const QImage &image, ZXing::BarcodeFormats formats) {
+
+    m_decoder.process(image, SCodes::toZXingFormat(SCodes::SBarcodeFormat::Basic));
+
+    continueProcessing();
+}
```

Comment:
The `*decoder` and `formats` variable seems to be unused

Reply:
Fixed